### PR TITLE
Allow move command to run on other OS besides Windows

### PR DESCRIPTION
**Pull request by scleaver**
Not everyone runs Windows lads ;)

### CODE REVIEW

**Comment by Piedone** [comment]
Thank you! At one point I did think about this, but then I figured Seth will do it if he needs it :D.

File: Lombiq.Gulp.Extensions.csproj [hunk 1]
```diff
@@ -15,7 +15,9 @@
   
   <Target Name="InitCommonNodeModules" BeforeTargets="ResolveReferences" Inputs="package.json" Outputs="$(InitCommonNodeModulesStampFile)">
     <Exec Command="npm install" />
-    <Exec Command="move node_modules ../.." Condition="Exists('node_modules')" />
+    <Exec Command="move node_modules ../.." Condition="'$(OS)' == 'Windows_NT' And Exists('node_modules')" />
+    <Exec Command="mv node_modules ../.." Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::OSX)))' And Exists('node_modules')" />
+    <Exec Command="mv node_modules ../.." Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Linux)))' And Exists('node_modules')" />
```
Comment (Piedone) [inline Lombiq.Gulp.Extensions.csproj]: Did you try the Move task? https://docs.microsoft.com/en-us/visualstudio/msbuild/move-task That might work as well in a single command, without having to run different commands for each OS.

If that doesn't work then please move the OSX and Linux ones into one command, the two OS checks OR-d together.

Comment (Piedone) [inline Lombiq.Gulp.Extensions.csproj]:
```suggestion
    <Exec Command="move node_modules ../.." Condition="'$(OS)' == 'Windows_NT' AND Exists('node_modules')" />
    <Exec Command="mv node_modules ../.." Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::OSX)))' AND Exists('node_modules')" />
    <Exec Command="mv node_modules ../.." Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Linux)))' AND Exists('node_modules')" />
```

**Review by Piedone — CHANGES_REQUESTED** [review]

**Comment by Piedone** [comment]
Did you have a chance to look at this @scleaver?

**Comment by scleaver** [comment]
@Piedone - please check this will work for Windows as it works in Mac (although it leaves a folder structure behind in the node_modules)... particularly check that my change to InitCommonNodeModulesStampFile mets your approval. `../../node_modules/.install-stamp` seemed a bit fragile to me and I thought it would be better if it used the solution dir instead `$(SolutionDir)\node_modules\.install-stamp`

File: Lombiq.Gulp.Extensions.csproj [hunk 2]
```diff
@@ -7,10 +7,11 @@
 
   <ItemGroup>
     <None Remove="node_modules\**" />
+    <FilesToMove Include="$(ProjectDir)\node_modules\**"/>
```
Comment (Piedone) [inline Lombiq.Gulp.Extensions.csproj]: Both `ProjectDir` and `SolutionDir` include a trailing slash (see https://docs.microsoft.com/en-us/cpp/build/reference/common-macros-for-build-commands-and-properties?view=msvc-160) so they shouldn't be followed by one.

**Review by Piedone — CHANGES_REQUESTED** [review]

**Comment by scleaver** [comment]
Apologies - should be good now.

**Comment by scleaver** [comment]
The move task failed in a Linux build - currently testing to see why.

**Comment by scleaver** [comment]
@Piedone - I could not get `<Move SourceFiles="@(FilesToMove)" DestinationFolder="$(SolutionDir)node_modules\%(RecursiveDir)" />` to work within a Linux build pipeline. It kept reporting after the move task that the "The file "/home/vsts/work/1/s/node_modules/.install-stamp" cannot be created. Could not find a part of the path '/home/vsts/work/1/s/node_modules/.install-stamp'." - it seems that for some reason the move task does not work correctly so the time-stamp can not be created but there is no additional information on the move task in the log so I don't know why.

The move task works fine on my Mac though so not sure.

I moved back to `<Exec Command="mv node_modules $(SolutionDir)" Condition="'$(OS)' != 'Windows_NT' AND Exists('node_modules')" />` and it went smoothly. The move task moves individual files not the folders so the empty folders remain  in the project folder however the `%(RecursiveDir)` worked to maintain the folder structure when moving.

I'll leave it with you as to what to do here... I still think it is improved over the original. It just depends on whether you want to solve the move task issue or not in order to use it.

**Comment by Piedone** [comment]
I see. Well, the joys of hooking into MSBuild internals, been there, done that :).

This will be fine for now, thank you. Just please be so kind and add a short comment to the commands why we're using them and not `Move`.

**Comment by Piedone** [comment]
Thank you. I had some further fixes in the https://github.com/Lombiq/Gulp-Extensions/commits/issue/FINI-506. I'll merged this there and if everything is alright, then later that to dev.

I can't change the base branch of this PR though so I don't think I can make GitHub think it is merged... So I'm just closing this and watch https://github.com/Lombiq/Gulp-Extensions/pull/9 for getting merged.

**Pull request closed without merging**